File: Auxiliary/gclib/examples/cs/examples/examples/Remote_Client.cs
/** @addtogroup cs_examples
  * @{
  */

/*! \file remote_client.cs
*
* Function calls for the Remote Client Example Project.
* <br><br>For VB.NET, see definition in file remote_client.vb
*/
using System;

namespace examples
{
    public static partial class Examples
    {
        /** @addtogroup cs_examples
        * @{
        */
        /// <summary>
        /// Accepts user input to publish to list and connect to available servers.
        /// </summary>
        /// <returns>The success status or error code of the function.</returns>
        /// <remarks>
        /// |Key 	  | Usage										   |
        /// |:-------:|:----------------------------------------------:|
        /// | q    	  | Quit										   |
        /// | s  	  | List available servers on then network		   |
        /// | h  	  | List available hardware on the current server  |
        /// | 0-9  	  | Connect to server instance by number		   |
        /// | l  	  | Connect back to local server				   |
        /// See remote_client_example.cs for an example.</remarks>
        /*! For VB.NET, see definition in file remote_client.vb */
        public static int Remote_Client()
        {
            gclib gclib = new gclib();

            bool loop = true;
            string[] servers_list = Array.Empty<string>();

            Console.WriteLine("<s> List available servers on the network\n" +
                            "<h> List available hardware on currently connected server\n" +
                            "<0-9> Enter numbers 0-9 to connect to a server by index\n" +
                            "<l> Set active server back to local server\n" +
                            "<q> Quit");

            while (loop)
            {
                char input = Console.ReadKey(true).KeyChar;

                if(input == 'q')
                {
                    loop = false;
                }
                else if(input == 's')
                {
                    Console.WriteLine("Available Servers:");
                    servers_list = gclib.GListServers();
                    Print_Servers_List(servers_list);
                }
                else if(input >= '0' && input <= '9')
                {
                    int index = input - '0';
                    if(servers_list.Length > 0 && index < servers_list.Length)
                    {
                        gclib.GSetServer(servers_list[index]);
                        Console.WriteLine("Server set to: " + servers_list[index]);
                    }
                }
                else if(input == 'l')
                {
                    gclib.GSetServer("Local");
                    Console.WriteLine("Server set to: Local");
                }
                else if(input == 'h')
                {
                    string[] addresses = gclib.GAddresses();

                    foreach(string address in addresses)
                    {
                        Console.WriteLine(address);
                    }               
                }
            }

            return GALIL_EXAMPLE_OK;
        }

        private static void Print_Servers_List(string[] servers_list)
        {
            if(servers_list.Length == 0)
            {
                Console.WriteLine("none");
            }
            else
            {
                for(int i = 0; i < servers_list.Length; i++)
                {
                    Console.WriteLine("<" + i + "> " + servers_list[i]);
                }
            }
        }
        /** @}*/
    }
}
/** @}*/

File: Auxiliary/gclib/examples/vb/examples/examples/Remote_Client.vb
Partial Public Module Examples
    ''' <summary>
    ''' Demonstrates various uses of GListServers() and GSetServer()
    ''' </summary>
    ''' <returns>The success status Or error code of the function.</returns>
    ''' <remarks>See remote_client_example.cs for an example.</remarks>
    Function Remote_Client() As Integer
        Dim gclib = New Gclib()

        Console.WriteLine("<s> List available servers on the network" + vbNewLine +
                            "<h> List available hardware on currently connected server" + vbNewLine +
                            "<0-9> Enter numbers 0-9 to connect to a server by index" + vbNewLine +
                            "<l> Set active server back to local server" + vbNewLine +
                            "<q> Quit")

        Dim loop_bool = True
        Dim servers_list As String() = Array.Empty(Of String)

        While loop_bool
            Dim input As Char = Console.ReadKey(True).KeyChar

            If input = "q" Then
                loop_bool = False
            ElseIf input = "s" Then
                Console.WriteLine("Available Servers:")
                servers_list = gclib.GListServers()
                Print_Servers_List(servers_list)
            ElseIf input >= "0" And input <= "9" Then
                Dim index As Integer = Convert.ToInt16(input) - Convert.ToInt16("0"c)
                If servers_list.Length > 0 And index < servers_list.Length Then
                    gclib.GSetServer(servers_list(index))
                    Console.WriteLine("Server set to: " + servers_list(index))
                End If
            ElseIf input = "l" Then
                gclib.GSetServer("Local")
                Console.WriteLine("Server set to: Local")
            ElseIf input = "h" Then
                Dim addresses As String() = gclib.GAddresses()

                For Each address As String In addresses
                    Console.WriteLine(address)
                Next
            End If
        End While

        Return Examples.GALIL_EXAMPLE_OK
    End Function

    Private Sub Print_Servers_List(servers_list As String())
        If servers_list.Length = 0 Then
            Console.WriteLine("none")
        Else
            For i As Integer = 0 To servers_list.Length - 1
                Console.WriteLine("<" + i.ToString() + "> " + servers_list(i))
            Next
        End If

    End Sub
End Module
